R/edition.R

```diff
@@ -58,14 +62,16 @@ edition_name <- function(x) {
 #' @param .env Environment that controls scope of changes. For expert use only.
 #' @keywords internal
 local_edition <- function(x, .env = parent.frame()) {
-  stopifnot(is.numeric(x) && length(x) == 1)
+  stopifnot(is_zap(x) || (is.numeric(x) && length(x) == 1))
   old <- edition_set(x)
   withr::defer(edition_set(old), envir = .env)
 }
 
 edition_set <- function(x) {
   if (is_zap(x)) {
+    old <- env_get(testthat_env, "edition", default = 2L)
```

Comment:
Are sure it makes sense to set a default here?

Reply:
TBH, didn't think that much about it.

In my tests, if no default is needed, we can simplify the whole thing to

```
edition_set <- function(x) {
  env_poke(testthat_env, "edition", x)
}
```

This still works with `zap()` and keeps a consistent return type. How does that sound?

Reply:
Yeah, that makes sense. I probably wrote this for an earlier version of `env_poke()`

Reply:
Thanks!